homeassistant/components/device_tracker/googlehome.py

```diff
@@ -5,91 +5,80 @@
 https://home-assistant.io/components/device_tracker.googlehome/
 """
 import logging
+from datetime import timedelta
 
-import voluptuous as vol
+from homeassistant.components.device_tracker import DeviceScanner
+from homeassistant.components.googlehome import (
+    CLIENT, DOMAIN as GOOGLEHOME_DOMAIN, NAME)
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.util import slugify
 
-import homeassistant.helpers.config_validation as cv
-from homeassistant.helpers.aiohttp_client import async_get_clientsession
-from homeassistant.components.device_tracker import (
-    DOMAIN, PLATFORM_SCHEMA, DeviceScanner)
-from homeassistant.const import CONF_HOST
+DEPENDENCIES = ['googlehome']
 
-REQUIREMENTS = ['ghlocalapi==0.3.5']
+DEFAULT_SCAN_INTERVAL = timedelta(seconds=10)
 
 _LOGGER = logging.getLogger(__name__)
 
-CONF_RSSI_THRESHOLD = 'rssi_threshold'
 
-PLATFORM_SCHEMA = vol.All(
-    PLATFORM_SCHEMA.extend({
-        vol.Required(CONF_HOST): cv.string,
-        vol.Optional(CONF_RSSI_THRESHOLD, default=-70): vol.Coerce(int),
-    }))
-
-
-async def async_get_scanner(hass, config):
-    """Validate the configuration and return an Google Home scanner."""
-    scanner = GoogleHomeDeviceScanner(hass, config[DOMAIN])
-    await scanner.async_connect()
-    return scanner if scanner.success_init else None
+async def async_setup_scanner(hass, config, async_see, discovery_info=None):
+    """Validate the configuration and return a Google Home scanner."""
+    scanner = GoogleHomeDeviceScanner(hass, hass.data[CLIENT],
```

Comment:
Add a gaurd clause that checks `if discovery_info is None` and `return False` if so.

Reply:
This will only be `None` if a user adds this as a platform and not configure with component?
If that is the case should I also add an error message?
```
_LOGGER.error("To use this platform you need to configure the 'googlehome' component.")
```

Reply:
You could add a warning but we usually don't do that.